parl/framework/tests/test_algorithm.py

```diff
@@ -0,0 +1,155 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import paddle.fluid as fluid
+from paddle.fluid.layer_helper import LayerHelper
+import parl.layers as layers
+from parl.framework.net import Model, Algorithm, create_algorithm_func, Feedforward
+import numpy as np
+import unittest
+
+
+class TestModel1(Model):
+    def __init__(self, dims):
+        super(TestModel1, self).__init__()
+        self.dims = dims
+        self.fc = layers.fc(dims)
+
+    def get_input_specs(self):
+        return [("sensor", dict(shape=[self.dims]))]
+
+    def get_action_specs(self):
+        return [("continuous_action", dict(shape=[self.dims]))]
+
+    def perceive(self, inputs, states):
+        hidden = self.fc(input=inputs.values()[0])
+        return dict(hidden=hidden), states
+
+
+class TestAlgorithm1(Algorithm):
+    def __init__(self, num_dims):
+        super(TestAlgorithm1, self).__init__(gpu_id=-1)
+        self.mlp = Feedforward([layers.fc(num_dims) for _ in range(1)])
+
+    def _predict(self, policy_states):
+        return dict(continuous_action=self.mlp(policy_states.values()[0]))
+
+    def _learn(self, policy_states, next_policy_states, actions, rewards):
+        return dict(cost=rewards.values()[0] - rewards.values()[0])
+
+
+class TestAlgorithm(unittest.TestCase):
+    def test_sync_paras_in_one_program(self):
+        """
+        Test case for copying parameters
+        """
+
+        algorithm_func = create_algorithm_func(
```

Comment:
Still feel that using model object and algorithm object is better. The user can control what to share himself.

Reply:
Maybe I can write a clone() function that creates new parameter names for the LayerFunc class.